pkg/kubectl/cmd/run.go

```diff
@@ -274,9 +286,43 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 				ResourceNames(mapping.Resource, name).
 				Flatten().
 				Do()
-			return ReapResult(r, f, cmdOut, true, true, 0, -1, false, mapper)
+			err = ReapResult(r, f, cmdOut, true, true, 0, -1, false, mapper)
+			if err != nil {
+				return err
+			}
+		}
+
+		if !waitForExitCode {
+			return nil
+		}
+
+		switch pod.Status.Phase {
+		case api.PodSucceeded:
+			return nil
+		case api.PodFailed:
+			if len(pod.Status.ContainerStatuses) == 0 || pod.Status.ContainerStatuses[0].State.Terminated == nil {
+				return fmt.Errorf("pod %s/%s failed with unknown exit code", pod.Namespace, pod.Name)
+			}
+			if len(pod.Status.ContainerStatuses) > 1 {
```

Comment:
``kubectl run`` should create a pod with one container only. So this condition will never be true (most likely). However, as the code just prints messages and one can never be sure of non-existence of some black magic, it does not hurt to keep it there.